enhancements/etcd/automated-backups-no-config.md

```diff
@@ -0,0 +1,271 @@
+
+To improve the user's experience, this work proposes taking etcd backup using default config from day one, without additional configuration from the user.
+
+### Goals
+
+* Backups should be taken without configuration after cluster installation from day 1.
+* Backups are saved to a default location on each master node's disk using `hostPath` volume.
+* Backups are taken according to a default schedule.
+* Backups are being maintained according to a default retention policy.
+
+### Non-Goals
+
+* Save cluster backups to remote cloud storage (e.g. S3 Bucket).
+  - This could be a future enhancement or extension to the API.
+  - Having backups in an independent storage from OCP cluster is crucial for Disaster Recovery scenarios.
+* Automate cluster restoration.
+* Provide automated backups for non-self hosted architectures like Hypershift.
+
+## Proposal
+
+To enable automated etcd backups of an Openshift cluster from Day 1, without user supplied configuration, this work proposes a default etcd backup option.
+
+The default backup option utilizes a sidecar container within each etcd pod. The container is responsible for maintaining backups within a `hostPath` volume on each master node's disk.
+
+This approach guarantees have backups across all master nodes which is vital in disaster recovery scenarios.
+
+### User Stories
+
+* As a cluster administrator, I would like OCP cluster backup to be taken without configuration from day one.
+* As a cluster administrator, I would like to schedule recurring OCP cluster backups so that I have a recent cluster state to recover from in the event of quorum loss (i.e. losing a majority of control-plane nodes).
+
+### API Extensions
+
+No [API](https://github.com/openshift/enhancements/blob/master/enhancements/etcd/automated-backups.md#api-extensions) changes are required.
+
+### Topology Considerations
+TBD
+#### Hypershift / Hosted Control Planes
+TBD
+#### Standalone Clusters
+TBD
+#### Single-node Deployments or MicroShift
+TBD
+### Workflow Description
+TBD
+
+## Design Details
+
+A sidecar container is being deployed within each etcd pod in Openshift cluster in order to take backups according to default configuration using `config.openshift.io/v1alpha1 Backup`.
+
+To distinguish default _no-config_ backups from user configured counterparts, a `config.openshift.io/v1alpha1 Backup` CR's name must be `default`.
+
+Initially, the sidecar container within etcd pods is not running (i.e. disabled). However, creating a `Backup` CR with `name=default` enables the container.
+
+Once enabled, the backups are being taken on each master node according to the provided `schedule` within the CR. Moreover, the backups are being maintained according to the supplied retention policy. 
+
+The default CR's configuration could be overridden by the user according to their preference.
+
+The backups are being stored within a `hostPath` volume. The volume is a directory within each master node's file system.
```

Comment:
How are files expected to be named?

Reply:
```
oc rsh -c etcd-backup-server -n openshift-etcd pod/etcd-ip-10-0-109-64.ec2.internal 
sh-5.1# ls -l /var/backup/etcd/
total 0
drwxr-xr-x. 2 root root 96 Aug 26 15:10 2024-08-26_151000
drwxr-xr-x. 2 root root 96 Aug 26 15:15 2024-08-26_151500
drwxr-xr-x. 2 root root 96 Aug 26 15:20 2024-08-26_152000
```

see the comments on this [PR](https://github.com/openshift/cluster-etcd-operator/pull/1325#issuecomment-2310546816), it has a test for the whole work

Reply:
it follows this naming https://github.com/openshift/cluster-etcd-operator/blob/0b7f488071c585a131483ab4e3dae050faeed32c/test/e2e/backup_test.go#L671-L675

Reply:
ah, okay, so it has a date which is useful. I think its worth adding the IP too, so that when admins collect backups from different nodes they wouldn't have them mixed. Not sure if this is important though (the DB should be consistent across other nodes too, so the backups are probably interchangeable)

Reply:
so the backups are taken by each backup pod from the local etcd member on the same node 

 I can add the ip-address, it is not an issue, 

Wdyt @hasbro17 @tjungblu